### ISSUE TITLE
Can TensorRT-LLM support the modified QWenAttention

### ISSUE THREAD

**Issue by Hukongtao** [NVIDIA/TensorRT-LLM#1037]
As the title describes, I slightly modified the QWenAttention.
Before:
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/9ee6d300-5f92-489d-8022-cdf467f9acb2)
After:
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/37b7fab1-7937-47dc-9b94-99644bbc37db)
self.is_causal = False

I use tensorrt-llm to accelerate the modified Qwen model, but there will be a problem of inconsistent numerical accuracy.
I want to know if there is a way to support the modified QWenAttention.

**Comment by Hukongtao** [comment]
For attention_mask_type,  I tried both causal and bidirectional, but the results didn't make any difference.
https://github.com/NVIDIA/TensorRT-LLM/blob/3d56a445e8ebf888e78be638faf6beec0a78f3c2/tensorrt_llm/models/qwen/model.py#L133

**Comment by Hukongtao** [comment]
> For attention_mask_type, I tried both causal and bidirectional, but the results didn't make any difference.
> 
> https://github.com/NVIDIA/TensorRT-LLM/blob/3d56a445e8ebf888e78be638faf6beec0a78f3c2/tensorrt_llm/models/qwen/model.py#L133

What is the difference between AttentionMaskType.causal and AttentionMaskType.bidirectional? Why use different types but the hidden_state is the same?

**Comment by Hukongtao** [comment]
Hi I solved my problem by change the true to false:
https://github.com/NVIDIA/TensorRT-LLM/blob/3d56a445e8ebf888e78be638faf6beec0a78f3c2/cpp/tensorrt_llm/plugins/gptAttentionCommon/gptAttentionCommon.cpp#L1439
